CarPG/Assets/Scripts/PlayerDeath.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+namespace UnityStandardAssets.Vehicles.Car
+{
+    public class PlayerDeath : MonoBehaviour
+    {
+        public bool dead;
+        // Start is called before the first frame update
+        void Start()
+        {
+            dead = false;
+
+        }
+
+        // Update is called once per frame
+        void Update()
+        {
+            if (dead == false)
+            {
+                if (GetComponent<Damagable>() != null)
+                {
+                    float h = GetComponent<Damagable>().health;
+                    if (h <= 0)
+                    {
+                        dead = true;
+                        Death();
+                    }
+                }
+            }
+        }
```

Comment:
Instead of checking the health of the Character every frame you should take out the update function and make a Die() function that calls Death(). Or you could just rename Death() to Die() tbh. Damagable will call Die() on all components when it dies and by using the method we'll save on performance significantly.